scene/2d/animated_sprite_2d.cpp

```diff
@@ -122,7 +122,7 @@ void AnimatedSprite2D::_validate_property(PropertyInfo &property) const {
 			}
 
 			property.hint_string += String(E->get());
-			if (animation == E) {
+			if (animation == String(E->get())) {
```

Comment:
`animation` is a `StringName`, so it would be better to compare to `E->get()` (also `StringName`) instead of casting to `String`.